readthedocs/search/views.py

```diff
@@ -43,84 +45,117 @@ def elastic_search(request, project_slug=None):
     :param project_slug: Sent when the view is a project search
     """
 
-    request_type = None
-    if project_slug:
-        queryset = Project.objects.protected(request.user)
-        project_obj = get_object_or_404(queryset, slug=project_slug)
-        request_type = request.GET.get('type', 'file')
-
-    user_input = UserInput(
-        query=request.GET.get('q'),
-        type=request_type or request.GET.get('type', 'project'),
-        project=project_slug or request.GET.get('project'),
-        version=request.GET.get('version', LATEST),
-        language=request.GET.get('language'),
-        role_name=request.GET.get('role_name'),
-        index=request.GET.get('index'),
-    )
-    search_facets = collections.defaultdict(
-        lambda: ProjectSearch,
-        {
-            'project': ProjectSearch,
-            'file': PageSearch,
-        }
-    )
-
-    results = []
-    facets = {}
+    http_method_names = ['get']
 
-    if user_input.query:
-        filters = {}
+    def _get_project(self, project_slug):
+        queryset = Project.objects.protected(self.request.user)
+        project = get_object_or_404(queryset, slug=project_slug)
+        return project
```

Comment:
I wonder if this function could be taken from a Mixin somewhere? Seems like we already have this defined somewhere.

Reply:
We have it, but the project_slug is taken from the url (kwargs), not from the query params like we do here

Reply:
I think I'm wrong here, I thought this was the api view, here we get it from the kwargs